### PR TITLE
Namespace whitelisting in helm chart without clusterRole

### PR DESCRIPTION
**Pull request by PaulFarver**
When not using a clusterRole, flux gets restricted to the namespace it is deployed in.

This pull request adds an option to let flux monitor and manipulate resources in other namespaces, without being deployed with a clusterrole

This is done by creating roles and rolebindings in all namespaces listed in a new value `allowedNamespaces`

### CODE REVIEW

File: chart/flux/README.md [hunk 1]
```diff
@@ -202,10 +202,11 @@ The following tables lists the configurable parameters of the Flux chart and the
...
+| `allowedNamespaces`                               | `[]`                                                 | Allow flux to manage resources in the specified namespaces. The namespace flux is deployed in will always be included
 | `serviceAccount.create`                           | `true`                                               | If `true`, create a new service account
 | `serviceAccount.name`                             | `flux`                                               | Service account to be used
 | `serviceAccount.annotations`                      | ``                                                   | Additional Service Account annotations
-| `clusterRole.create`                              | `true`                                               | If `false`, Flux and the Helm Operator will be restricted to the namespace where they are deployed
+| `clusterRole.create`                              | `true`                                               | If `false`, Flux and the Helm Operator will be restricted to the namespaces given in `allowedNamespaces` and the namespace where they are deployed
```
Comment (stefanprodan) [inline chart/flux/README.md]: Helm Operator is no longer deployed with this chart, please remove it from the readme and values.

Thanks

File: chart/flux/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 appVersion: "1.17.0"
-version: 1.0.0
+version: 1.1.0
```
Comment (stefanprodan) [inline chart/flux/Chart.yaml]: Please undo this change, the version bump happens at release time.

Comment (PaulFarver) [reply chart/flux/Chart.yaml]: Right. Fixed now

File: chart/flux/README.md (hunk 1 above)
Comment (PaulFarver) [reply chart/flux/README.md]: I updated the documentation as well as the comment in the values file

File: chart/flux/templates/deployment.yaml [hunk 3]
```diff
@@ -175,7 +175,7 @@ spec:
           {{- end }}
           args:
           {{- if not .Values.clusterRole.create }}
-          - --k8s-allow-namespace={{ .Release.Namespace }}
+          - --k8s-allow-namespace={{ join "," (append .Values.allowedNamespaces .Release.Namespace) }}
```
Comment (2opremio) [inline chart/flux/templates/deployment.yaml]: Uhm, this may result in `.Release.Namespace` being duplicated which may cause problems (or inefficiencies). Would you mind changing the type of `NewCluster()`'s `allowedNamespaces` argument and `Cluster.allowedNamespaces` from `[]string` to a set (`map[string]struct{}`)?

Comment (PaulFarver) [reply chart/flux/templates/deployment.yaml]: Good catch. I didn't think of that 👍
